### PR TITLE
Feature/359 profile files

### PR DESCRIPTION
**Pull request by mitzimorris**
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and open-source license: see below

#### Summary

Add argument 'save_profile' to methods for sampling, optimization, and variational inference.
Didn't add to `generate_quantities` but can add there too for consistency.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):  Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

File: cmdstanpy/cmdstan_args.py [hunk 1]
```diff
@@ -497,7 +497,7 @@ def validate(self, chains=None) -> None:  # pylint: disable=unused-argument
     # pylint: disable=unused-argument
     def compose(self, idx: int, cmd: List) -> str:
         """
-        Compose CmdStan command for method-specific non-default arguments.
+        Compose CmdStan command for method<-specific non-default arguments.
```
Comment (ahartikainen) [inline cmdstanpy/cmdstan_args.py:500]:
```suggestion
        Compose CmdStan command for method-specific non-default arguments.
```

**Review by ahartikainen — COMMENTED** [review]
Looks good. There are couple binary files that needs to be removed.

**Comment by mitzimorris** [comment]
somehow compiled binaries from the unit tests were checked into GitHub.  OK to remove as part of this PR?

**Comment by mitzimorris** [comment]
otherwise will just do a cleanup on develop

**Comment by syclik** [comment]
Please do! Thank you!

**Comment by syclik** [comment]
(Please remove them on this PR.... sorry, I just realized that the response was ambiguous.)

**Pull request merged**